src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/ReleaseDataFileServiceTests.cs

```diff
@@ -2943,10 +2962,179 @@ public async Task UploadAsBulkZip()
+
+            var dataArchiveValidationService = new Mock<IDataArchiveValidationService>(Strict);
+            var fileUploadsValidatorService = new Mock<IFileUploadsValidatorService>(Strict);
+            var privateBlobStorageService = new Mock<IPrivateBlobStorageService>(/*Strict*/); // Failing mock verification on UploadStream
 
+            dataArchiveValidationService
+                .Setup(s => s.IsValidZipFile(archive))
+                .ReturnsAsync([]);
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                dataArchiveValidationService
+                    .Setup(s => s.ValidateBulkDataArchiveFiles(
+                        releaseVersion.Id,
+                        archive))
+                    .ReturnsAsync(new List<ArchiveDataSetFile>
+                    {
+                        archiveFile1, archiveFile2,
+                    });
+
+                //privateBlobStorageService.Setup(mock =>
+                //    mock.UploadStream(
+                //        PrivateReleaseTempFiles,
+                //        It.Is<string>(path => path.Contains(FilesPath(releaseVersion.Id, FileType.Data, It.IsAny<Guid>()))),
+                //        It.IsAny<Stream>(),
+                //        ContentTypes.Csv,
+                //        null,
+                //        default
+                //    )).Returns(Task.CompletedTask);
+
+                var service = SetupReleaseDataFileService(
+                    contentDbContext: contentDbContext,
+                    dataArchiveValidationService: dataArchiveValidationService.Object,
+                    fileUploadsValidatorService: fileUploadsValidatorService.Object,
+                    privateBlobStorageService: privateBlobStorageService.Object
+                );
+
+                var result = (await service
+                    .ValidateAndUploadBulkZip(releaseVersion.Id, archive, default))
+                    .AssertRight();
+
+                MockUtils.VerifyAllMocks(
+                    dataArchiveValidationService,
+                    fileUploadsValidatorService//,
+                    /*privateBlobStorageService*/);
+
+                Assert.Equal(2, result.Count);
+
+                Assert.Equal("One", result[0].Title);
+                Assert.Equal("one.csv", result[0].DataFilename);
+                Assert.Equal("one.meta.csv", result[0].MetaFilename);
+
+                Assert.Equal("Two", result[1].Title);
+                Assert.Equal("two.csv", result[1].DataFilename);
+                Assert.Equal("two.meta.csv", result[1].MetaFilename);
+            }
+        }
+
+        // TODO: Refactor 2x duplicate functions in DataArchiveValidationServiceTests
+        // Possibly in MockFormTestUtils
+        private static IFormFile CreateFormFileFromResource(string fileName, string? newFileName = null)
```

Comment:
Found another :)

There is another lower down too, but I won't leave a comment for any more I find